articles/views.py

```diff
@@ -32,3 +35,124 @@ def delete(self, request: Request, *args: Any, **kwargs: Any) -> Response:
             {"message": "Article deleted successfully"},
             status=status.HTTP_204_NO_CONTENT,
         )
+
+
+class CommentListView(generics.GenericAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    queryset = Article.objects.all()
+    renderer_classes = (JSONRenderer,)
+
+    def post(self, request: Request, slug: str) -> Response:
+        """
+        Allows authenticated users can add comments on articles
+        Args:
+            slug: this a slug for a particular article
+        Returns:
+            code: The return 201 created for success
+        """
+        self.author = get_object_or_404(Profile, user=request.user)
+        self.article = get_object_or_404(Article, slug=slug)
+        start_index = validate_index(request.data.get("highlight_start"), slug)
+        end_index = validate_index(request.data.get("highlight_end"), slug)
+        if start_index and end_index:
+            selected = (
+                [int(start_index), int(end_index)]
+                if start_index < end_index
+                else [int(end_index), int(start_index)]
+            )
+            highlight_text = str(self.article.body[selected[0] : selected[1]])
+            request.data["highlight_text"] = highlight_text
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(author=self.author, article=self.article)
+        return Response(
+            {
+                "message": "Comment added successfully",
+                "comment": serializer.data,
+            },
+            status=status.HTTP_201_CREATED,
+        )
+
+    def get(self, request: Request, slug: str) -> Response:
```

Comment:
Most of this functionality is handled for you with the RetrieveAPIView and pagination